docs/gallery/plot_term_set.py

```diff
@@ -0,0 +1,143 @@
+   As mentioned prior, the URI needs to be to the terms; this is to allow the URI to later be coupled
+   with the source id for the term to create a valid link to the term source page.
+3. The schema uses LinkML enumerations to list all the possible terms. Currently, users will need to
+   manually outline the terms within the enumeration's permissible values.
+
+For a clear example, please refer to example_term_set.yaml within the tutorial gallery.
+"""
+######################################################
+# Creating an instance of the TermSet class
+# ----------------------------------------------------
+from hdmf.common import DynamicTable, VectorData
+import os
+
+try:
+    dir_path = os.path.dirname(os.path.abspath(__file__))
+    yaml_file = os.path.join(dir_path, 'example_term_set.yaml')
+except NameError:
+    dir_path = os.path.dirname(os.path.abspath('.'))
+    yaml_file = os.path.join(dir_path, 'gallery/example_term_set.yaml')
+
+######################################################
+# Viewing TermSet values
+# ----------------------------------------------------
+# :py:class:`~hdmf.term_set.TermSet` has methods to retrieve terms. The :py:func:`~hdmf.term_set.TermSet:view_set`
+# method will return a dictionary of all the terms and the corresponding information for each term.
+# Users can index specific terms from the :py:class:`~hdmf.term_set.TermSet`. LinkML runtime will need to be installed.
+# You can do so by first running ``pip install linkml-runtime``.
+from hdmf.term_set import TermSet
+terms = TermSet(term_schema_path=yaml_file)
+print(terms.view_set)
+
+# Retrieve a specific term
+terms['Homo sapiens']
+
+######################################################
+# Validate Data with TermSet
+# ----------------------------------------------------
+# :py:class:`~hdmf.term_set.TermSet` has been integrated so that :py:class:`~hdmf.container.Data` and its
+# subclasses support a term_set attribute. By having this attribute set, the data will be validated
+# and all new data will be validated.
+data = VectorData(
+    name='species',
+    description='...',
+    data=['Homo sapiens'],
+    term_set=terms)
+
+######################################################
+# Validate on append with TermSet
+# ----------------------------------------------------
+# As mentioned prior, when the term_set attribute is set all new data is validated. This is true for both
+# append and extend methods.
+data.append('Ursus arctos horribilis')
+data.extend(['Mus musculus', 'Myrmecophaga tridactyla'])
+
+######################################################
+# Validate Data in a DynamicTable with TermSet
+# ----------------------------------------------------
+# Validating data with :py:class:`~hdmf.common.table.DynamicTable` is determined by which columns were
+# initialized with the term_set attribute set. The data is validated when the columns are created and not
+# when set as columns to the table.
```

Comment:
```suggestion
# initialized with the term_set attribute set. The data is validated when the columns are created or
# modified. Since adding the columns to a DynamicTable does not modify the data, validation is
# not being performed at that time.
```